src/test/java/com/google/api/codegen/testdata/go_example_library.baseline

```diff
@@ -28,185 +28,267 @@ func ExampleNewClient() {
     ctx := context.Background()
     opts := []gax.ClientOption{ /* Optional client parameters. */ }
     c, err := library.NewClient(ctx, opts...)
-    _, _ = c, err // Handle error.
+    if err != nil {
+        // TODO: Handle error.
+    }
+    // TODO: Use client.
+    _ = c
 }
 
 func ExampleClient_CreateShelf() {
     ctx := context.Background()
     c, err := library.NewClient(ctx)
-    _ = err // Handle error.
+    if err != nil {
+        // TODO: Handle error.
+    }
 
     req := &google_example_library_v1.CreateShelfRequest{ /* Data... */ }
     var resp *google_example_library_v1.Shelf
     resp, err = c.CreateShelf(ctx, req)
-    _, _ = resp, err // Handle error.
+    if err != nil {
+        // TODO: Handle error.
+    }
+    // TODO: Use resp.
+    _ = resp
 }
 
 func ExampleClient_GetShelf() {
     ctx := context.Background()
     c, err := library.NewClient(ctx)
-    _ = err // Handle error.
+    if err != nil {
+        // TODO: Handle error.
+    }
 
     req := &google_example_library_v1.GetShelfRequest{ /* Data... */ }
     var resp *google_example_library_v1.Shelf
     resp, err = c.GetShelf(ctx, req)
-    _, _ = resp, err // Handle error.
+    if err != nil {
+        // TODO: Handle error.
+    }
+    // TODO: Use resp.
+    _ = resp
 }
 
 func ExampleClient_ListShelves() {
     ctx := context.Background()
     c, err := library.NewClient(ctx)
-    _ = err // Handle error.
+    if err != nil {
+        // TODO: Handle error.
+    }
 
     req := &google_example_library_v1.ListShelvesRequest{ /* Data... */ }
     it := c.ListShelves(ctx, req)
     var resp *google_example_library_v1.Shelf
     for {
         resp, err = it.Next()
```

Comment:
var resp to inside the for loop.

Also, this code leaks "err" to the outside scope. That's partly why I don't like "var resp" here.

We don't tell people the types for the discovery doc samples. We should be consistent. Also, they can just look it up in the godoc for "ListShelves".